django/contrib/postgres/search.py

```diff
@@ -240,6 +242,65 @@ def as_sql(self, compiler, connection, function=None, template=None):
         return sql, extra_params + params
 
 
+class SearchHeadline(Func):
+    function = 'ts_headline'
+    output_field = TextField()
+    options = {}
+    VALID_OPTIONS = (
+        'StartSel',
+        'StopSel',
+        'MaxWords',
+        'MinWords',
+        'ShortWord',
+        'HighlightAll',
+        'MaxFragments',
+        'FragmentDelimiter',
+    )
+
+    def __init__(self, expression, query, config=None, **extra):
+        if not hasattr(query, 'resolve_expression'):
+            query = SearchQuery(query)
+        self.config = SearchConfig(config) if config else None
+        options = extra.get('options') or {}
+        self.options = {**self.options, **options}
+        invalid_options = [option for option in self.options.keys() if option not in self.VALID_OPTIONS]
+        if invalid_options:
+            raise ValueError("Unknown options key(s) '%s'." % ', '.join(invalid_options))
+
+        super().__init__(expression, query, **extra)
+
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        resolved = super().resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        if self.config:
+            resolved.config = self.config.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        return resolved
+
+    def as_sql(self, compiler, connection, function=None, template=None, arg_joiner=None, **extra_context):
+        config_params, options_params = [], []
+        extra_context = {}
+        if template is None:
+            template = '%(function)s(%(config)s%(expressions)s%(options)s)'
+            extra_context.update({
+                'config': '',
+                'options': '',
+            })
+            if self.config:
+                config_sql, config_params = compiler.compile(self.config)
+                extra_context['config'] = '{}, '.format(config_sql)
+            if self.options:
+                options = []
+                for option, value in self.options.items():
+                    pg_value = psycopg2.extensions.adapt(value).getquoted().decode()
+                    options.append('%s=%s' % (option, pg_value))
+                options_params = [', '.join(options)]
```

Comment:
append instead of creating a new list

```suggestion
                options_params.append(', '.join(options))
```

All off the above could also be reduced to

```python
 options_params.append(', '.join(
    '%s=%s' % (option, psycopg2.extensions.adapt(value).getquoted().decode())
    for option, value in options.items()
))

Reply:
Much nicer, thanks!